src/procedures/proc_bfs.c

```diff
@@ -200,12 +200,12 @@ static SIValue *Proc_BFS_Step
 		}
 
 		res = GxB_Vector_Iterator_next(iter);
-		id = GxB_Vector_Iterator_getIndex(iter);
+		if(res == GrB_SUCCESS) id = GxB_Vector_Iterator_getIndex(iter);
```

Comment:
I think this condition can be removed if we change the while loop to a for loop:
```c
GrB_Vector_nvals(&nvals, bfs_ctx->nodes);
for (GrB_Index i = 0; i < nvals; i++) {
...
    id = GxB_Vector_Iterator_getIndex(iter);
    ASSERT(res == GrB_SUCCESS);
    id = GxB_Vector_Iterator_getIndex(iter);
```